web/src/components/lineage/components/node/Node.tsx

```diff
@@ -62,11 +76,11 @@ class Node extends React.Component<NodeProps> {
             <circle
               style={{ cursor: 'pointer' }}
               r={RADIUS}
-              fill={theme.palette.common.white}
+              fill={runStateToNodeColor(job.latestRun)}
               stroke={
                 selectedNode === node.label
-                  ? theme.palette.primary.main
-                  : theme.palette.secondary.main
+                  ? theme.palette.common.white
+                  : runStateToNodeColor(job.latestRun)
```

Comment:
You've got the case where you're selected and failing, you can just move all this logic up to your function above.

Reply:
I think mixing both a ternary and a switch is too confusing.

Reply:
Ok, good point. I'll clean things up 👍